### PR TITLE
Provisional PR #2

### PR DESCRIPTION
**Pull request by Oaphi**
This is a second provisional PR with a bunch of fixes/enhancements to the codebase, including:

- package with auto-generated types for SE API (I am the creator and maintainer)
- improvements to the `chatIDToSiteID` signature: with `client` type fixed, it must allow `string | number`
- switch to gereric type parameters for `addCommand` and `deleteCommand` to allow subtypes of `Client` to be passed as the second parameter of the `cb` member of the `Command` interface
- tsconfig inheritance for platform-specific tsconfigs to keep the config DRY and improve maintainability of shared config
- bump of fetch-cookie to 2.0.5 to ensure a stable version is used

### CODE REVIEW

File: packages/bot/src/Bot.ts [hunk 1]
```diff
@@ -150,11 +150,17 @@ export class Bot extends events.EventEmitter {
         this.clientFunctions.push(func);
     }
 
-    addCommand(cmd: Command) {
+    /**
+     * @summary registers a {@link Command}
```
Comment (jbis9051) [inline packages/bot/src/Bot.ts:154]: I think comment unnecessary

File: packages/bot/src/types/CallbackTypes.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
 import { Client } from '../Client';
 import { Message } from '../models/Message';
 
-export type MessageHandler<T = void> = (msg: Message, client: Client) => T;
+export type MessageHandler<T = void, U extends Client = Client> = (msg: Message, client: U) => T;
```
Comment (jbis9051) [inline packages/bot/src/types/CallbackTypes.ts:4]: Does the generic T make any sense?

File: packages/bot/src/Bot.ts (hunk 1 above)
Comment (Oaphi) [reply packages/bot/src/Bot.ts:154]: I am used to JSDoc'ing all methods/function signatures, but the comment is actually incomplete and thus looks off, I forgot the `@param cmd {@link Command} to register`. However, if you prefer not to add JSDoc, can remove, ofc

Comment (jbis9051) [reply packages/bot/src/Bot.ts:154]: I think ts documents most things fine, don't think it's needed

File: packages/bot/src/types/CallbackTypes.ts (hunk 2 above)
Comment (Oaphi) [reply packages/bot/src/types/CallbackTypes.ts]: While I was making the change, I noticed that there was, indeed, a case where having a generic type parameter for the return type of the function signature made sense - some of the `MessageHandler`s return `boolean` and are used as validators. However, I didn't find any other return types, so simply settled on `boolean | void | Promise<void>` - that was enough to cover all cases. 

Had to apply a small fix for `RegisterShutdownScript`, though, as it used a concise body and thus ended up returning `Promise<Promise<void>[]>`. On a related note, there was a strange thing about it: you had `Object.values()` and a `// @ts-ignore` comment whereas `Object.entries()` was supposed to be used judging from the context. I fixed it, but not sure if there was a very specific reason in doing so.

File: packages/bot/src/Bot.ts (hunk 1 above)
Comment (Oaphi) [reply packages/bot/src/Bot.ts]: No problem with that (just my default preference at play) - removed those JSDoc comments

File: packages/plugins/src/plugins/stat.ts [hunk 3]
```diff
@@ -1,5 +1,10 @@
-import { PluginFunction, Client, PermissionType } from '@chatbot/bot';
+import { PermissionType, PluginFunction } from '@chatbot/bot';
+import { SOClient } from "@chatbot/so/SOClient";
 
+/**
+ * @summary adds a 'stat' command for user stats
+ * @param bot bot instance
```
Comment (jbis9051) [inline packages/plugins/src/plugins/stat.ts:6]: not needed

**Pull request merged**